core/src/main/java/tc/oc/pgm/structure/StructureDefinition.java

```diff
@@ -48,10 +50,22 @@ public Vector getOrigin() {
     return this.origin.clone();
   }
 
+  @Override
   public Bounds getBounds() {
     if (bounds == null) {
       this.bounds = region.getBounds();
     }
     return bounds;
   }
+
+  @Override
+  public void place(BlockVector offset) {
+    throw new UnsupportedOperationException("Cannot call place from StructureDefinition!");
+  }
+
+  @Override
+  public void placeAbsolute(Match match, Vector vector) {
+    WorldStructure worldStructure = match.getFeatureContext().get(getId(), WorldStructure.class);
+    worldStructure.placeAbsolute(match, vector);
+  }
```

Comment:
i'm not sure structure definition should implement structure, often times there's a clear cut difference between the match-scoped object and the map-scoped object (definition). 

Sometimes the lines are a bit blurrier for objects that don't have separate match and map scopes, (eg: filters, actions, or regions, who are stateless), but do note that stateful objects (eg: VariableDefinition & Variable, or TeamFactory & Team, or CoreFactory & Core) do have clearly defined separate definitions and impls, and often times a way to map (VariableDefinition#getVariable, or GoalDefinition#getGoal)

You can bridge the gap with a `Structure getStructure(Match match)` method